nimare/stats.py

```diff
@@ -136,24 +136,22 @@ def null_to_p(test_value, null_array, tail="two"):
     P-values are clipped based on the number of elements in the null array.
     Therefore no p-values of 0 or 1 should be produced.
     """
+    test_value = np.atleast_1d(test_value)
+    # TODO: this runs in N^2 time; is there a more efficient alternative?
+    p = np.array([stats.percentileofscore(null_array, v, "rank") for v in test_value])
```

Comment:
If you would rather it take `N^2 space`, I think I translated the rank portion of the [percentileofscore function](https://github.com/scipy/scipy/blob/v1.5.4/scipy/stats/stats.py#L1942-L2029)

```
def percentileofscore(a, scores):
    n = len(a)
    a_repeat = np.repeat(a[:, np.newaxis], scores.shape[0], axis=1)
    left = np.count_nonzero(a_repeat < scores, axis=0)
    right = np.count_nonzero(a_repeat <= scores, axis=0)
    # percentage
    return (right + left + (right > left)) * 50.0/n
```

Don't know if that would be feasible for the size of the arrays we need.

Reply:
Here is a test for this function to make sure it does the same thing as scipy:

```
from nimare import stats
from scipy.stats import percentileofscore as scipy_percentileofscore

def test_percentileofscore():
    a = np.arange(1, 101)
    scores = np.arange(0, 50)
    nimare_p = stats.percentileofscore(a, scores)
    scipy_p = np.array([scipy_percentileofscore(a, score) for score in scores])
    assert np.all(nimare_p == scipy_p)
```

Reply:
Good thought, but not really feasible—we're looking at ~200k voxels for typical analyses, and a 200k x 200k matrix isn't going to fit in memory on most machines (and even if it did, it would be very slow). I think a more efficient solution will probably use something like [numpy.searchsorted](https://numpy.org/doc/stable/reference/generated/numpy.searchsorted.html) to do the heavy lifting.

Feel free to open a PR (or add commits to this one) if you like—though it's not a very high priority, as `null_to_p` is usually only invoked once per analysis. So right now it seems to add a constant 2 - 3 seconds to any `fit()` call that uses MNI 2 mm space, but it doesn't get any worse for bigger analyses (including more permutations).